src/blog/01_26_propositions_and_predicates.md

```diff
@@ -0,0 +1,164 @@
+
+# Introduction to Predicates
+
+
+## Recap of Propositions:
+
+For the purpose of propositional logic, **propositions** are (almost) atomic black-boxes:
+They have _names_ and _definitions_, but their names are just atomic symbols in the logic syntax,
+and their definitions are _outside_ the logic syntax (usually these are statements in a human language).
+
+The set of all propositions, together with the special symbols ⊤ and ⊥,
+are the atomic symbols of the syntax of **formulas**.
+The connectives will be familiar from their roles as boolean operators: ¬, ∧ ,∨, →, _etc_.
+Thus, formulas are recursive trees with connectives as nodes and atomic symbols as leaves. 
+
+There are two truth values: 1 and 0.
+(These are not part of the syntax of formulas.)
+These can go by other names, commonly _"true"_ and _"false"_.
+The properties and isomorphisms of [Boolean Algebra](https://en.wikipedia.org/wiki/Boolean_algebra)
+are all good fun, but for now we can simply take **truth values** ≜ {1, 0}.
+
+The operator **⟦·⟧** maps formula to truth values; _e.g._ ⟦⊤⟧=1.
+(This example is actually part of the definition of ⟦·⟧;
+⊤ and ⊥ are given meaning respectively as an atomic tautology and an atomic contradiction.)
+Although it's often left implicit, it technically has a subscript argument,
+either an **assignment** (a set of (_proposition_, _truth value_) pairs)
+or a pair (_KB_, σ) of a **knowlege base** (a list of statements of the same form as proposition definitions)
+and a **storage** of proposition definitions.
+In either case, ⟦·⟧₋ can be implemented in terms of lookups and if-then-else statements. 
+
+
+## Predicates as Relations:
+
+Consider the category _B_ of truth values, and whatever category _S_ of nouns/subjects.
+A single proposition (_e.g._ _p_≜`"Six is prime."`) can be represented as a pair of form 
+(_s_∈_S_, _R_) where _R_ is a [relation](https://en.wikipedia.org/wiki/Binary_relation) over _S_ and _B_
```

Comment:
In class I've used the notation (_s_, _b_) $\in$ _R_ where _s_ and _b_ are elements of the sets _S_ and _R_. If you could  note that what you have written here is equivalent, that would help other students. I just worry about having different syntax and want to ensure that students following what I have been presenting in class understand how it maps to what they are reading here.